src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -136,6 +136,13 @@ public IEnumerable<SonarQubeIssue> GetSuppressedIssues(string projectGuid, strin
 
         public IEnumerable<SonarQubeIssue> GetAllSuppressedIssues()
         {
+            // Block the call while the cache is being built.
+            // If the task has already completed then this will return immediately
+            // (e.g. on subsequent calls)
+            // If we time out waiting for the initial fetch then we won't suppress any issues.
+            // We'll try to fetch the issues again when the timer elapses.
+            this.initialFetch?.Wait(MillisecondsToWaitForInitialFetch);
```

Comment:
Previously this was always called AFTER the list was already fetched. Now it's possible that it's called while the first fetch is being executed, in which case we need to block.
The other method has a lot of tests that check that the call is blocked but it wrong to just copy-paste them. I think this class could use some refactoring but that's obviously out of scope for now.

Reply:
It looks like it would be really easy to turn this into an `async` method...
Still, now isn't the time to do it. Let's keep it in mind as we add new services.